src/main/java/com/baidu/hugegraph/manager/AuthBackupRestoreManager.java

```diff
@@ -0,0 +1,574 @@
+import com.baidu.hugegraph.structure.constant.HugeType;
+import com.baidu.hugegraph.util.E;
+import com.baidu.hugegraph.util.JsonUtil;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+public class AuthBackupRestoreManager extends BackupRestoreBaseManager {
+
+    private static final String AUTH_BACKUP_RESTORE = "auth-backup-restore";
+    private static final int NO_CONFLICT = 0;
+
+    private AuthRestoreStrategy strategy;
+    private String initPassword;
+    /*
+     * The collection of id relationships of users, groups and targets
+     * is the basic data of belong and accesses。
+     */
+    private Map<String, String> idsMap;
+    private Map<String, User> usersByName;
+    private Map<String, Group> groupsByName;
+    private Map<String, Target> targetsByName;
+    private Map<String, Belong> belongsByName;
+    private Map<String, Access> accessesByName;
+    private List<AuthManager> authManagers;
+
+    public AuthBackupRestoreManager(ToolClient.ConnectionInfo info) {
+        super(info, AUTH_BACKUP_RESTORE);
+    }
+
+    public void init(SubCommands.AuthBackup authBackup) {
+        this.retry(authBackup.retry());
+        this.directory(authBackup.directory(), authBackup.hdfsConf());
+        this.ensureDirectoryExist(true);
+        this.authManagers = Lists.newArrayList();
+    }
+
+    public void init(SubCommands.AuthRestore authRestore) {
+        this.retry(authRestore.retry());
+        this.directory(authRestore.directory(), authRestore.hdfsConf());
+        this.ensureDirectoryExist(false);
+        this.strategy = AuthRestoreStrategy.fromName(authRestore.strategy());
+        this.initPassword(authRestore.types(), authRestore.initPassword());
+        this.idsMap = Maps.newHashMap();
+        this.usersByName = Maps.newHashMap();
+        this.groupsByName = Maps.newHashMap();
+        this.targetsByName = Maps.newHashMap();
+        this.belongsByName = Maps.newHashMap();
+        this.accessesByName = Maps.newHashMap();
+        this.authManagers = Lists.newArrayList();
+    }
+
+    public void backup(List<HugeType> types) {
+        try {
+            this.doAddAuths(types);
+            this.doBackup();
+        } catch (Throwable e) {
+            throw e;
+        } finally {
+            this.shutdown(this.type());
+        }
+    }
+
+    private void doBackup() {
+        for (AuthManager authManager : this.authManagers) {
+            authManager.backup();
+        }
+    }
+
+    public void restore(List<HugeType> types) {
+        List<HugeType> sortedHugeTypes = this.sortListByCode(types);
+        try {
+            this.doAddAuths(sortedHugeTypes);
+            this.doRestore();
+        } catch (Throwable e) {
+            throw e;
+        } finally {
+            this.shutdown(this.type());
+        }
+    }
+
+    private void doRestore() {
+        for (AuthManager authManager : this.authManagers) {
+            authManager.checkConflict();
```

Comment:
collect all the conflicts and print error message if exist any one:
"Conflicts encountered before restoring users and stop this action"

Reply:
address it